src/language/tsLanguageService.ts

```diff
@@ -0,0 +1,215 @@
+      target: ts.ScriptTarget.Latest,
+      moduleResolution: ts.ModuleResolutionKind.Classic,
+      experimentalDecorators: false,
+    };
+    let currentTextDocument = TextDocument.create('init', 'javascript', 1, '');
+
+    // Create the language service host to allow the LS to communicate with the host.
+    const host: ts.LanguageServiceHost = {
+      getCompilationSettings: () => compilerOptions,
+      getScriptFileNames: () => [currentTextDocument.uri],
+      getScriptKind: () => {
+        return ts.ScriptKind.JS;
+      },
+      getScriptVersion: (fileName: string) => {
+        if (fileName === currentTextDocument.uri) {
+          return String(currentTextDocument.version);
+        }
+        return '1';
+      },
+      getScriptSnapshot: (fileName: string) => {
+        let text = '';
+        if (fileName === currentTextDocument.uri) {
+          text = currentTextDocument.getText();
+        } else {
+          text = this._loadLibrary(fileName);
+        }
+        return {
+          getText: (start, end) => text.substring(start, end),
+          getLength: () => text.length,
+          getChangeRange: () => undefined,
+        };
+      },
+      getCurrentDirectory: () => '',
+      getDefaultLibFileName: () => GLOBAL_CONFIG_LIBRARY_NAME,
+      readFile: (path: string): string | undefined => {
+        if (path === currentTextDocument.uri) {
+          return currentTextDocument.getText();
+        }
+        return this._loadLibrary(path);
+      },
+      fileExists: (): boolean => false,
+      directoryExists: (): boolean => false,
+    };
+
+    // Create the language service files.
+    const jsLanguageService = ts.createLanguageService(host);
+
+    return {
+      // Return a new instance of the language service.
+      getLanguageService(jsDocument: TextDocument): ts.LanguageService {
+        currentTextDocument = jsDocument;
+        return jsLanguageService;
+      },
+      getCompilationSettings() {
+        return compilerOptions;
+      },
+      dispose() {
+        jsLanguageService.dispose();
+      },
+    };
+  }
+
+  doSignatureHelp(
+    document: TextDocument,
+    position: Position
+  ): Promise<SignatureHelp | null> {
+    const jsDocument = TextDocument.create(
+      document.uri,
+      'javascript',
+      document.version,
+      document.getText()
+    );
+    const jsLanguageService = this._host.getLanguageService(jsDocument);
+    const signHelp = jsLanguageService.getSignatureHelpItems(
+      jsDocument.uri,
+      jsDocument.offsetAt(position),
+      undefined
+    );
+
+    if (signHelp) {
+      const ret: SignatureHelp = {
+        activeSignature: signHelp.selectedItemIndex,
+        activeParameter: signHelp.argumentIndex,
+        signatures: [],
+      };
+      signHelp.items
+        .map((item) => {
+          const hasInt8Array = item.prefixDisplayParts.filter(
+            (prefix) => prefix.text !== 'Int8Array'
+          );
```

Comment:
I think there might be room for a bit of readability improvement here – `hasInt8Array` sounds like it would be a boolean, but it seems like it’s an array of items that are specifically not referring to `Int8Array`, so maybe there’s a better variable name? And finally, I think it’s definitely worth leaving a comment about why `Int8Array` is excluded (and why only it and none of the other typed arrays)